libvast/builtins/operators/summarize.cpp

```diff
@@ -650,28 +680,28 @@ class plugin final : public virtual operator_plugin<summarize_operator> {
       };
     }
     auto config = configuration{};
-    for (const auto& [output, function_name, arguments] :
+    for (const auto& [output, function_name, argument] :
          std::get<0>(parsed_aggregations)) {
       configuration::aggregation new_aggregation{};
-      if (!plugins::find<aggregation_function_plugin>(function_name)) {
+      auto const* function
+        = plugins::find<aggregation_function_plugin>(function_name);
+      if (!function) {
         return {
           std::string_view{f, l},
-          caf::make_error(ec::syntax_error, fmt::format("invalid aggregation "
-                                                        "name: '{}'",
+          caf::make_error(ec::syntax_error, fmt::format("invalid "
+                                                        "aggregation "
+                                                        "function `{}`",
                                                         function_name)),
         };
       }
-      new_aggregation.function_name = function_name;
-      new_aggregation.input_extractors = arguments;
+      new_aggregation.function = function;
+      new_aggregation.input = argument;
       new_aggregation.output
-        = (output)
-            ? *output
-            : fmt::format("{}({})", function_name, fmt::join(arguments, ","));
+        = (output) ? *output : fmt::format("{}({})", function_name, argument);
       config.aggregations.push_back(std::move(new_aggregation));
     }
     config.group_by_extractors = std::move(std::get<1>(parsed_aggregations));
```

Comment:
More of a question: The linked issue also talks about making `[by …]` optional. Such an implementation could entirely bypass the grouping mechanism so it only needs to iterate over the data once. Did you want intend to implement that as part of this PR, or is it a follow-up one?

Reply:
Done: 098d3acc148c2dd570afdf1ba7da5ff2d074d2d9. This implementation still goes through the grouping mechanism, but with an empty data vector. This is probably the easiest way to implement and maintain this.

Reply:
Nice, that's a really small diff for this!